pkg/compactor/blocks_cleaner.go

```diff
@@ -391,3 +408,49 @@ func (c *BlocksCleaner) cleanUserPartialBlocks(ctx context.Context, partials map
 		level.Info(userLogger).Log("msg", "deleted partial block marked for deletion", "block", blockID)
 	}
 }
+
+// applyUserRetentionPeriod marks blocks for deletion which have aged past the retention period.
+func (c *BlocksCleaner) applyUserRetentionPeriod(ctx context.Context, idx *bucketindex.Index, userID string, userBucket *bucket.UserBucketClient, userLogger log.Logger) {
```

Comment:
Nit: it feels like this method is taking too many reduntant parameters. It could create userLogger and userBucket on its own, since it already has userID.

Alternatively we could remove userID and pass it retention directly, since we're already passing userBucket and logger -- no more need for userID.

Reply:
I mainly followed the pattern of cleanUserPartialBlocks here:

```
func (c *BlocksCleaner) cleanUserPartialBlocks(ctx context.Context, partials map[ulid.ULID]error, idx *bucketindex.Index, userBucket *bucket.UserBucketClient, userLogger log.Logger) {
```

For consistency, I think I prefer the option of passing retention, and removing `userID`. What do you think?